### PR TITLE
Add form component for editing deployments

### PR DESCRIPTION
**Pull request by divyanshiGupta**
**Fixes:** https://issues.redhat.com/browse/ODC-5453

In this PR I have added a form component that will be used for editing the deployments.
This form will be integrated into edit-deployment feature in this story https://issues.redhat.com/browse/ODC-5454.
During integration I will be refactoring a few components like `HealthCheckProbes`, `ImageSearchSection`, `SourceSecretSelector` and make them generic so that they can be used for this form as well and we don't have duplicate any code. Any other remaining issues in the form (if any) will be fixed in the upcoming story.

/kind feature

### CODE REVIEW

File: frontend/packages/dev-console/src/components/edit-deployment/AdvancedSection.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { useTranslation } from 'react-i18next';
+import ScalingSection from '../import/advanced/ScalingSection';
+import FormSection from '../import/section/FormSection';
+import ProgressiveList from '../progressive-list/ProgressiveList';
+import ProgressiveListItem from '../progressive-list/ProgressiveListItem';
+import PauseRolloutsSection from './PauseRolloutsSection';
+
+const AdvancedSection: React.FC = () => {
+  const { t } = useTranslation();
+  const [visibleItems, setVisibleItems] = React.useState([]);
```
Comment (abhinandan13jan) [inline frontend/packages/dev-console/src/components/edit-deployment/AdvancedSection.tsx:11]: Do we need a type here?

File: frontend/packages/dev-console/src/components/edit-deployment/EditDeploymentFormEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import * as React from 'react';
```
Comment (abhinandan13jan) [inline frontend/packages/dev-console/src/components/edit-deployment/EditDeploymentFormEditor.tsx:2]: nit: move this above all

File: frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/RecreateStrategy.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { InputField } from '@console/shared/src';
+import { TextInputTypes } from '@patternfly/react-core';
+import * as React from 'react';
```
Comment (abhinandan13jan) [inline frontend/packages/dev-console/src/components/edit-deployment/deployment-strategy/RecreateStrategy.tsx:3]: nit: sort imports

**Review by rohitkrai03 — APPROVED** [review]
/approve
/lgtm

Code for all the new form sections and components looks good to me. We can get this merged in since the integration story will handle all the parts for connecting this form.

**Pull request merged**